Adding UMI consensus flow to the pipeline (#134)
nf-cmgg/preprocessing: README.md

The pipeline is built using Nextflow, a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It comes with docker containers making installation trivial and results highly reproducible.

> The pipeline also supports Unique Molecular Identifier (UMI) data. If your samplesheet includes a `umi_type` column (`seq` or `readname`), UMI-aware preprocessing is enabled automatically; rows with `umi_type=none` are processed as usual.

Reviewer: ```suggestion
The pipeline also supports Unique Molecular Identifier (UMI) data. If your samplesheet includes a `umi_type` column (`seq` or `readname`), UMI-aware preprocessing is enabled automatically. Rows with no `umi_type` specified will be processed as non-UMI sequencing data.
```

I thought this was a bit confusing

![metro map](docs/images/metro_map.png)

UMI processing (only for rows with `umi_type`):
- Extract UMI from read sequence (`seq`) or read name (`readname`)
- Group reads by UMI (fgbio GroupReadsByUmi)
- Call molecular consensus (fgbio CallMolecularConsensusReads) and filter (fgbio FilterConsensusReads)
- Re-align filtered consensus reads with BWA-MEM (`-Y`), then sort/index

> 

Reviewer: Can you guys also add this to the metro map if you have some time left? You can find tutorials for this here: https://nf-co.re/docs/guidelines/graphic_design/overview
